doc/adr/0011-tsp-analysis-api.md

```diff
@@ -8,124 +8,124 @@ Proposed
-    GET /analysis/types
-        returns a map typeId -> list of analysis descriptor of existing analyses on server: analysisId, name, description, parameters
-    GET /analysis/types/{typeId}
-        returns a list of analysis descriptors for given typeId
-    POST /analysis/types/{typeId}
-        Upload an analysis definition for given typeId
-        Returns a new analysis descriptor with unique analysisID
-    PUT /analysis/types/{typeId}/{analysisId}
-        Update an analysis
-    GET /analysis/types/{typeId}/{analysisId}
-        Returns the analysis descriptor for given analysisId
-    DELETE /analysis/types/{typeId}/{analysisId}
-        Delete an analysis
-
-#### Analysis type descriptor
-
-The analysis type descriptor describes the type of external analysis. Different types have different syntax and hence back-end implementation. This descriptor is used to distinquish the different external analysis. The trace server implementation will provide the list of type descriptors on client requests.
+The trace viewer currently is able to visualize trace data provided by a trace server over the trace server protocol (TSP). The Trace Compass server has some built-in analysis view for that. However, it is not possible to side-load analysis and visualization descriptions over the TSP so that end-user can provide some custom views. The Trace Compass supports loading of data-driven analysis and views, e.g. XML driven views or in-and-out anlysis of the Trace Compass incubator project. In the Eclispe-based Trace Compass application, there exists UI primitive to load e.g. XML files or configure custom analysis. While the Trace Compass server has the capablility to understand these defintions, there is no way to side-load this definition over the TSP. This ADR will propose a configuration service within the TSP to facilitate these custom analysis. The proposed configuration service can be use to configure other server specific customization, e.g. custom trace parsers.
+
+### Global configuration service
+
+An analysis service for managing global configurations will be introduced. Global in this context means that the configuration definitions will be handled on the application level. 
+
+    GET /config/
+        returns a list of configuration source types: typeId, name, description, scope, expected query parameter keys ("path" for file path)
+    GET /config/types
+        returns a map typeId -> list of configuration instances configured on the server for a given configuration source type: configId, name, description, parameters
+    GET /config/types/{typeId}
+        returns a list of configuration instances for a given typeId
+    POST /config/types/{typeId}
+        Upload an configuration definition for given typeId
+        Returns a new configuration descriptor with unique configuration ID
+    PUT /config/types/{typeId}/{configId}
+        Update a configuration
+    GET /config/types/{typeId}/{configId}
```

Comment:
As discussed offline, we are now proposing to change the endpoint paths to:
/config/types/{typeId} for operations on configuration types
/config/types/{typeId}/configs/{configId} for operations on configuration instances

Reply:
True, that makes sense. GET /config/types/{typeId} currently returns a list of configurations instead of a specific configuration source type. So, I'll use GET /config/types/{typeId} retrieving a single configuration source type and GET /config/types/{typeId}/configs/ for the list of configurations per typeId. The other configuration endpoints will follow:

GET    /config/types/{typeId}/configs                      List of configurations   
POST   /config/types/{typeId}/configs                    Create a new configuration
GET    /config/types/{typeId}/configs/{configId}      List a single configuration
PUT    /config/types/{typeId}/configs/{configId}      Update a single configuration
DELETE /config/types/{typeId}/configs/{configId}  Delete a single configuration